internal/bminventory/inventory.go

```diff
@@ -1668,6 +1676,107 @@ func (b *bareMetalInventory) UpdateClusterNonInteractive(ctx context.Context, pa
 	return b.v2UpdateClusterInternal(ctx, params, NonInteractive)
 }
 
+func checkPlatformWrongParamsInput(platform *models.Platform, userManagedNetworking *bool) error {
+	if platform != nil && userManagedNetworking != nil {
+		userManagedNetworkingStatus := "enabled"
+		if !*userManagedNetworking {
+			userManagedNetworkingStatus = "disabled"
+		}
+
+		if (!*userManagedNetworking && *platform.Type == models.PlatformTypeNone) || (*userManagedNetworking && models.PlatformTypeBaremetal == *platform.Type) {
+			return common.NewApiError(http.StatusBadRequest, errors.Errorf("Can't set %s platform with user-managed-networking %s", *platform.Type, userManagedNetworkingStatus))
+		}
+	}
+
+	return nil
+}
+
+func getActualUpdateClusterPlatformParams(platform *models.Platform, userManagedNetworking *bool, cluster *common.Cluster) (*models.Platform, *bool, error) {
+	// case 5
+	if platform == nil && userManagedNetworking == nil {
+		return nil, nil, nil
+	}
+
+	// case 1
+	if err := checkPlatformWrongParamsInput(platform, userManagedNetworking); err != nil {
+		return nil, nil, err
+	}
+
+	if *cluster.Platform.Type == models.PlatformTypeBaremetal {
+
+		// case 2
+		if !swag.BoolValue(userManagedNetworking) && (platform == nil || *platform.Type == models.PlatformTypeBaremetal) {
+			return nil, nil, nil
+		}
+
+		// case 3
+		if (platform != nil && *platform.Type == models.PlatformTypeNone) || (swag.BoolValue(userManagedNetworking) && platform == nil) {
+			return &models.Platform{Type: common.PlatformTypePtr(models.PlatformTypeNone)}, swag.Bool(true), nil
+		}
+	}
+
+	if *cluster.Platform.Type == models.PlatformTypeNone {
+
+		// case 4
+		if (userManagedNetworking == nil || *userManagedNetworking) && (platform == nil || *platform.Type == models.PlatformTypeNone) {
+			return nil, nil, nil
+		}
+
+		// case 6
+		if !swag.BoolValue(userManagedNetworking) && (platform == nil || *platform.Type == models.PlatformTypeBaremetal) {
+			if cluster.CPUArchitecture != common.X86CPUArchitecture && !featuresupport.IsFeatureSupported(cluster.OpenshiftVersion, models.FeatureSupportLevelFeaturesItems0FeatureIDARM64ARCHITECTUREWITHCLUSTERMANAGEDNETWORKING) {
+				return nil, nil, common.NewApiError(http.StatusBadRequest, errors.New("disabling User Managed Networking or setting Bare-Metal platform is not allowed for clusters with non-x86_64 CPU architecture"))
+			}
```

Comment:
Consider moving this to another PR just in case it creates unexpected noise and might get reverted

Reply:
Without it `[It] Fail with non-x86_64 CPU architecture with 4.10` will fail